### PR TITLE
Use struct init shorthand when applicable in fill struct fields assist

### PR DESCRIPTION
**Pull request by nathanwhit**
This PR tweaks the fill struct fields assist to use the struct init shorthand when a local variable with a matching name and type is in scope.

For example:
```rust
struct Foo {
    a: usize,
    b: i32,
    c: char,
}

fn main() {
    let a = 1;
    let b = 2;
    let c = 3;
    let foo = Foo { <|> };
}
```
Before we would insert
```rust
Foo {
    a: (),
    b: (),
    c: (),
}
```
now we would insert
```rust
Foo {
    a,
    b,
    c: ()
}
```

### CODE REVIEW

File: crates/ide_diagnostics/src/handlers/missing_fields.rs [hunk 1]
```diff
@@ -54,9 +55,26 @@ fn fixes(ctx: &DiagnosticsContext<'_>, d: &hir::MissingFields) -> Option<Vec<Ass
...
-    for f in d.missed_fields.iter() {
+    let mut locals = FxHashMap::default();
+    ctx.sema.scope(field_list_parent.syntax()).process_all_names(&mut |name, def| {
+        if let hir::ScopeDef::Local(local) = def {
+            locals.insert(name.clone(), local);
+        }
+    });
+    let missing_fields = ctx.sema.record_literal_missing_fields(&field_list_parent);
+    for (f, ty) in missing_fields.iter() {
+        let field_expr = if let Some(local_candidate) = locals.get(&f.name(ctx.sema.db)) {
```
Comment (matklad) [inline crates/ide_diagnostics/src/handlers/missing_fields.rs:66]: Let's add a coverage mark here: https://docs.rs/cov-mark/1.1.0/cov_mark/

**Review by matklad — COMMENTED** [review]
Lovely :heart_eyes:, I've been wanting this since forever!

**Comment by matklad** [comment]
bors d+

**Comment by nathanwhit** [comment]
bors r=matklad

**Pull request merged**